[Bug]: Add Resource to Library dialog double installs and errors out

### Describe the Bug

When installing a new library via _File_ > _Add Resource to Library…_ the selected library is installed successfully, but then gets installed a second time, which presents a failure dialog to the user. The _Library_ window also presents a redundant version of the library with a blank name.

### To Reproduce

1. Open a fresh instance of MapTool
2. Go to _File_ > _Add Resource to Library…_
3. Go to the RPTools tab and select a library that you don't yet have installed.
4. Press the _Install_ button.
5. See a dialog show up saying "Error while downloading library". Dismiss this.
6. Look at the _Library_ window. See that the library was installed. Also see a blank entry at the top containing the same contents.

### Expected Behaviour

The library should only be downloaded once, and no blank entry should be added to the _Library_ window.

### Screenshots

This shows up after hitting _Install_:
<img width="267" height="154" alt="Image" src="https://github.com/user-attachments/assets/85e36f99-9307-4326-940b-0e8941c9c78f" />

And this is what the _Library_ window looks like, with the blank entry selected at the top:
<img width="436" height="70" alt="Image" src="https://github.com/user-attachments/assets/b78e733e-03d8-48a9-9e57-5fac4ab5fc98" />

### MapTool Info

1.18.5

### Desktop

Linux Mint 22.2

### Additional Context

This is what gets logged when trying to install:
````
2025-11-16 22:51:25.038 ERROR net.rptools.maptool.client.ui.addresource.AddResourceDialog - Error downloading library: java.io.FileNotFoundException: http://library.rptools.net/1.3/http://library.rptools.net/1.3/artpacks/CSP_Dungeons_to_Caves_Grid50.zip
java.io.FileNotFoundException: http://library.rptools.net/1.3/http://library.rptools.net/1.3/artpacks/CSP_Dungeons_to_Caves_Grid50.zip
	at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:486)
	at java.base/sun.net.www.protocol.http.HttpURLConnection$10.run(HttpURLConnection.java:2055)
	at java.base/sun.net.www.protocol.http.HttpURLConnection$10.run(HttpURLConnection.java:2050)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:571)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getChainedException(HttpURLConnection.java:2049)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1619)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1599)
	at net.rptools.maptool.client.RemoteFileDownloader.read(RemoteFileDownloader.java:71)
	at net.rptools.maptool.client.ui.addresource.AddResourceDialog$1.doInBackground(AddResourceDialog.java:292)
	at java.desktop/javax.swing.SwingWorker$1.call(SwingWorker.java:305)
	at java.base/java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:317)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java)
	at java.desktop/javax.swing.SwingWorker.run(SwingWorker.java:342)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.io.FileNotFoundException: http://library.rptools.net/1.3/http://library.rptools.net/1.3/artpacks/CSP_Dungeons_to_Caves_Grid50.zip
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1994)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1599)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getHeaderField(HttpURLConnection.java:3235)
	at java.base/java.net.URLConnection.getHeaderFieldLong(URLConnection.java:649)
	at java.base/java.net.URLConnection.getContentLengthLong(URLConnection.java:511)
	at java.base/java.net.URLConnection.getContentLength(URLConnection.java:495)
	at net.rptools.maptool.client.RemoteFileDownloader.read(RemoteFileDownloader.java:60)
	... 8 more
2025-11-16 22:51:25.040 INFO  net.rptools.maptool.client.MapTool - dialog.addresource.warn.couldnotload
````

--- a/src/main/java/net/rptools/maptool/client/swing/AbeillePanel.java
+++ b/src/main/java/net/rptools/maptool/client/swing/AbeillePanel.java
@@ -169,10 +169,8 @@ public class AbeillePanel<T> extends JPanel {
    */
   public void bind(T model) {
     if (this.model != null) {
-      // Jamz: Don't like this; the bind/unbind on open/close tracking. Binding can get locked on an
-      // exception rendering the dialog in a broken state.
+      log.error("Panel is already bound. Unbinding the old model.");
       unbind();
-      throw new IllegalStateException("Already bound exception");
     }
     this.model = model;
     Binder.bindContainer(model.getClass(), panel, UpdateTime.NEVER);

--- a/src/main/java/net/rptools/maptool/client/swing/GenericDialog.java
+++ b/src/main/java/net/rptools/maptool/client/swing/GenericDialog.java
@@ -19,8 +19,6 @@ import com.jidesoft.swing.*;
 import com.jidesoft.utils.PortingUtils;
 import java.awt.*;
 import java.awt.event.*;
-import java.util.ArrayList;
-import java.util.List;
 import javax.swing.*;
 import net.rptools.maptool.client.MapTool;
 import net.rptools.maptool.language.I18N;
@@ -28,15 +26,13 @@ import net.rptools.maptool.language.I18N;
 public class GenericDialog extends JDialog {
   public static final String AFFIRM = ButtonPanel.AFFIRMATIVE_BUTTON;
   public static final String DENY = ButtonPanel.CANCEL_BUTTON;
-  private Dimension _preferredSize = null;
   private boolean hasBeenShown;
   private String _dialogResult = ButtonPanel.CANCEL_BUTTON;
-  protected final Resizable _resizable;
+  private final Resizable _resizable;
   private final JComponent _contentPane = new JPanel();
   private JComponent _content = new JPanel();
   private final JScrollPane _scrollPane = new JScrollPane();
   private ScrollableButtonPanel _buttonPanel;
-  private boolean _usingAbeillePanel = false;
   private ActionListener _onCloseAction;
   private ActionListener _onShowAction;
 
@@ -227,7 +223,6 @@ public class GenericDialog extends JDialog {
 
   public void setContent(JComponent content) {
     this._content = content;
-    this._usingAbeillePanel = content instanceof AbeillePanel;
     this._scrollPane.setViewportView(content);
   }
 
@@ -259,24 +254,8 @@ public class GenericDialog extends JDialog {
         bounds.width - insets.left - insets.right, bounds.height - insets.top - insets.bottom);
   }
 
-  private List<Component> getAllComponents(Container c) {
-    List<Component> listOut = new ArrayList<>();
-    Component[] components = c.getComponents();
-    for (Component _c : components) {
-      listOut.add(_c);
-      if (_c instanceof Container c_) {
-        listOut.addAll(getAllComponents(c_));
-      }
-    }
-    return listOut;
-  }
-
   @Override
   public Dimension getPreferredSize() {
-    //    for (Component c : getAllComponents(this)) {
-    //      System.out.println(c.getName() + c.getPreferredSize());
-    //    }
-
     int scrollBarSize = UIManager.getDefaults().getInt("ScrollBar.width");
     Dimension frameSize = MapTool.getFrame().getSize();
     Dimension superPref = super.getPreferredSize();
@@ -355,12 +334,12 @@ public class GenericDialog extends JDialog {
   }
 
   public void closeDialog() {
-    if (this._usingAbeillePanel && ((AbeillePanel<?>) this._content).getModel() != null) {
+    if (this._content instanceof AbeillePanel<?> abeillePanel && abeillePanel.getModel() != null) {
       // wrap up any AbeillePanel commits and unbinding
       if (this.getDialogResult().equals(AFFIRM)) {
-        ((AbeillePanel<?>) _content).commit();
+        abeillePanel.commit();
       }
-      ((AbeillePanel<?>) _content).unbind();
+      abeillePanel.unbind();
     }
     // set off the onCloseAction if present
     if (this._onCloseAction != null) {

--- a/src/main/java/net/rptools/maptool/client/swing/GenericDialogFactory.java
+++ b/src/main/java/net/rptools/maptool/client/swing/GenericDialogFactory.java
@@ -19,23 +19,21 @@ import java.awt.event.ActionListener;
 import javax.swing.*;
 
 public class GenericDialogFactory {
-  GenericDialog delegate;
-
-  public GenericDialog getDialog() {
-    return delegate;
-  }
+  private final GenericDialog delegate;
 
   public GenericDialogFactory() {
     delegate = new GenericDialog();
   }
 
-  @SuppressWarnings("UnusedReturnValue")
+  public GenericDialog getDialog() {
+    return delegate;
+  }
+
   public GenericDialogFactory setDefaultButton(ButtonKind buttonKind) {
     delegate.setDefaultButton(buttonKind);
     return this;
   }
 
-  @SuppressWarnings("UnusedReturnValue")
   public GenericDialogFactory addButton(
       AbstractButton b, ActionListener l, Object constraints, int index) {
     b.addActionListener(l);
@@ -43,38 +41,31 @@ public class GenericDialogFactory {
     return this;
   }
 
-  @SuppressWarnings("UnusedReturnValue")
   public GenericDialogFactory addButton(AbstractButton b, ActionListener l, Object constraints) {
     return addButton(b, l, constraints, -1);
   }
 
-  @SuppressWarnings("UnusedReturnValue")
   public GenericDialogFactory addNonButton(Component c, Object constraints, int index) {
     delegate.addNonButton(c, constraints, index);
     return this;
   }
 
-  @SuppressWarnings("UnusedReturnValue")
   public GenericDialogFactory addNonButton(Component c, Object constraints) {
     return addNonButton(c, constraints, -1);
   }
 
-  @SuppressWarnings("UnusedReturnValue")
   public GenericDialogFactory addButton(ButtonKind buttonKind) {
     return addButton(buttonKind, null, null);
   }
 
-  @SuppressWarnings("UnusedReturnValue")
   public GenericDialogFactory addButton(ButtonKind buttonKind, Action action) {
     return addButton(buttonKind, action, null);
   }
 
-  @SuppressWarnings("UnusedReturnValue")
   public GenericDialogFactory addButton(ButtonKind buttonKind, ActionListener listener) {
     return addButton(buttonKind, null, listener);
   }
 
-  @SuppressWarnings("UnusedReturnValue")
   public GenericDialogFactory addButton(
       ButtonKind buttonKind, Action action, ActionListener listener) {
     delegate.addButton(buttonKind, action, listener);
@@ -89,67 +80,56 @@ public class GenericDialogFactory {
     return delegate.getButtonPanel().getOppositeButtonOrder();
   }
 
-  @SuppressWarnings("UnusedReturnValue")
   public GenericDialogFactory setButtonOrder(String buttonOrder) {
     delegate.setButtonOrder(buttonOrder);
     return this;
   }
 
-  @SuppressWarnings("UnusedReturnValue")
   public GenericDialogFactory setOppositeButtonOrder(String buttonOrder) {
     delegate.setOppositeButtonOrder(buttonOrder);
     return this;
   }
 
-  @SuppressWarnings("UnusedReturnValue")
   public GenericDialogFactory makeModal(boolean modal) {
     delegate.setModal(modal);
     return this;
   }
 
-  @SuppressWarnings("UnusedReturnValue")
   public GenericDialogFactory setCloseOperation(int closeOperation) {
     delegate.setDefaultCloseOperation(closeOperation);
     return this;
   }
 
-  @SuppressWarnings("UnusedReturnValue")
   public GenericDialogFactory createOkCancelButtons() {
     delegate.createOkCancelButtons();
     return this;
   }
 
-  @SuppressWarnings("UnusedReturnValue")
   public GenericDialogFactory onBeforeShow(ActionListener listener) {
     delegate.onBeforeShow(listener);
     return this;
   }
 
-  @SuppressWarnings("UnusedReturnValue")
   public GenericDialogFactory onBeforeClose(ActionListener listener) {
     delegate.onBeforeClose(listener);
     return this;
   }
 
-  @SuppressWarnings("UnusedReturnValue")
   public GenericDialogFactory setDialogTitle(String title) {
     delegate.setTitle(title);
     return this;
   }
 
-  @SuppressWarnings("UnusedReturnValue")
   public GenericDialogFactory setContent(JComponent content) {
     delegate.setContent(content);
     return this;
   }
 
-  @SuppressWarnings("UnusedReturnValue")
   public GenericDialogFactory setDialogResult(String string) {
     delegate.setDialogResult(string);
     return this;
   }
 
-  @SuppressWarnings("UnusedReturnValue")
   public GenericDialogFactory display() {
     delegate.showDialog();
     return this;

--- a/src/main/java/net/rptools/maptool/client/ui/addresource/AddResourceDialog.java
+++ b/src/main/java/net/rptools/maptool/client/ui/addresource/AddResourceDialog.java
@@ -48,8 +48,6 @@ import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
 public class AddResourceDialog extends AbeillePanel<AddResourceDialog.Model> {
-  private static final long serialVersionUID = -1709712124453405062L;
-
   private static final Logger log = LogManager.getLogger(AddResourceDialog.class);
 
   private static final String LIBRARY_URL = "http://library.rptools.net/1.3";
@@ -66,10 +64,7 @@ public class AddResourceDialog extends AbeillePanel<AddResourceDialog.Model> {
           .setDialogTitle(I18N.getText("action.addIconSelector"))
           .addButton(ButtonKind.CANCEL)
           .setCloseOperation(WindowConstants.DISPOSE_ON_CLOSE);
-  private Model model;
   private boolean downloadLibraryListInitiated;
-
-  private boolean install = false;
 
   public AddResourceDialog() {
     super(new AddRessourcesDialogView().getRootComponent());
@@ -77,30 +72,13 @@ public class AddResourceDialog extends AbeillePanel<AddResourceDialog.Model> {
     panelInit();
     dialogFactory
         .setContent(this)
-        .addButton(
-            ButtonKind.INSTALL,
-            e -> {
-              install = true;
-              if (commit()) {
-                dialogFactory.getDialog().closeDialog();
-              }
-            })
+        .addButton(ButtonKind.INSTALL)
         .setDefaultButton(ButtonKind.INSTALL);
   }
 
-  public boolean getInstall() {
-    return install;
-  }
-
   public void showDialog() {
-    model = new Model();
-    bind(model);
+    bind(new Model());
     dialogFactory.display();
-  }
-
-  @Override
-  public Model getModel() {
-    return model;
   }
 
   public JTextField getBrowseTextField() {
@@ -132,6 +110,7 @@ public class AddResourceDialog extends AbeillePanel<AddResourceDialog.Model> {
         .addChangeListener(
             e -> {
               // Hmmm, this is fragile (breaks if the order changes) rethink this later
+              var model = getModel();
               switch (tabPane.getSelectedIndex()) {
                 case 0 -> model.tab = Tab.LOCAL;
                 case 1 -> model.tab = Tab.WEB;
@@ -196,6 +175,7 @@ public class AddResourceDialog extends AbeillePanel<AddResourceDialog.Model> {
     // Add the resource
     final List<LibraryRow> rowList = new ArrayList<>();
 
+    var model = getModel();
     switch (model.getTab()) {
       case LOCAL -> {
         if (StringUtils.isEmpty(model.getLocalDirectory())) {

